ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -493,6 +493,72 @@ - (BOOL)isAccessibilityElement {
 @end
 
 
+#pragma mark - ORKFormItemConfirmTextCell
+
+@implementation ORKFormItemConfirmTextCell
+
+- (void)setSavedAnswers:(NSDictionary *)savedAnswers {
+    [super setSavedAnswers:savedAnswers];
+    
+    [savedAnswers addObserver:self
+                   forKeyPath:[self originalItemIdentifier]
+                      options:0
+                      context:nil];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary<NSString *,id> *)change context:(void *)context {
+    if ([keyPath isEqual:[self originalItemIdentifier]]) {
+        self.textField.text = nil;
+    }
+}
+
+- (BOOL)isAnswerValidWithString:(NSString *)string {
+    BOOL isValid = NO;
+    if (string.length > 0) {
+        NSString *originalItemAnswer = self.savedAnswers[[self originalItemIdentifier]];
+        if (!ORKIsAnswerEmpty(originalItemAnswer) && [originalItemAnswer isEqualToString:string]) {
+            isValid = YES;
+        }
+    }
+    return isValid;
+}
+
+- (NSString *)originalItemIdentifier {
+    ORKConfirmTextAnswerFormat *answerFormat = (ORKConfirmTextAnswerFormat *)self.formItem.answerFormat;
+    return [answerFormat.originalItemIdentifier copy];
+}
+
+- (void)dealloc {
+    [self.savedAnswers removeObserver:self forKeyPath:[self originalItemIdentifier]];
+}
+
+#pragma mark UITextFieldDelegate
+
+- (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string {
+    NSString *text = [textField.text stringByReplacingCharactersInRange:range withString:string];
+    [self ork_setAnswer:([self isAnswerValidWithString:text] ? text : @"")];
+
+    return YES;
+}
+
+- (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
+    [super textFieldShouldEndEditing:textField];
+    if (![self isAnswerValidWithString:textField.text] && textField.text.length > 0) {
+        textField.text = @"";
+        [self ork_setAnswer:textField.text];
+        [self showValidityAlertWithMessage:ORKLocalizedString(@"CONFIRM_PASSWORD_ERROR_MESSAGE", nil)];
+    }
+    return YES;
+}
+
+- (void)textFieldDidEndEditing:(UITextField *)textField {
+    self.editingHighlight = NO;
+    [self.delegate formItemCellDidResignFirstResponder:self];
```

Comment:
Yep good catch. I had to do this earlier when storing a boolean result, because otherwise it would try to store the textfield's text into a boolean result. But now that I switched back to storing the actual text, I can revert this as well!